core/corehttp/gateway_handler.go

```diff
@@ -523,6 +527,25 @@ func (i *gatewayHandler) serveFile(w http.ResponseWriter, req *http.Request, nam
 	http.ServeContent(w, req, name, modtime, content)
 }
 
+func (i *gatewayHandler) serveCBOR(w http.ResponseWriter, r *http.Request, n format.Node) {
+	w.Header().Set("Cache-Control", "public, max-age=29030400, immutable")
```

Comment:
This is invalid if DAG is fetched from anything other than `/ipfs/` like  `/ipns/` (not immutable). See [gateway_handler.go#L318-L319](https://github.com/ipfs/go-ipfs/blob/ea2bb34a7ad0704077180936f1ec3fe7ec25c6d2/core/corehttp/gateway_handler.go#L318-L319). 
This may require a bigger refactor, as we want to avoid separate code paths for unixfs and cbor.

Reply:
Good catch.